### DISCUSSION TITLE
how to  auto-starting the app on device boot？

### DISCUSSION THREAD

**Discussion by Youngzx88** [expo/expo#25630] [Archived: Questions] [answered]
I am developing an Android app using Expo. I would like to know how to implement the functionality of **auto-starting** the app on device boot. Is there any API available to listen for the "**BOOT_COMPLETE**D" event?

**Comment by ayberkanilatsiz** [comment]
@Youngzx88 Did you find any solution for it?

**Reply by ayberkanilatsiz** [reply, marked as answer]
@Youngzx88 i found solution.

run `npm expo prebuild` access native side.

https://stackoverflow.com/questions/53847965/android-boot-completed-notification-not-received-after-reboot

**Reply by Youngzx88** [reply]
I'm still at the requirements review level and haven't found a solution, but thanks for the solution。

**Reply by michalpospiech** [reply]
@Youngzx88 Did you manage to figure out a solution to auto start the application?

**Reply by Youngzx88** [reply]
So far, it appears that the only option is to convert the Expo project into a pure React Native project. @michalpospiech michalpospiech

**Comment by lukmccall** [comment]
You will need to add native code to your app because Expo does not directly support your use case. We don't have a ready package that handles that.
